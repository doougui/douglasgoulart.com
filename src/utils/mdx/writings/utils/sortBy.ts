import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
