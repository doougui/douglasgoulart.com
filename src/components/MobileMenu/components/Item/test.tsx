import { render, screen } from 'utils/tests';

import { Item } from '.';

describe('<Item />', () => {
  it('should render correctly', () => {
    const { container } = render(<Item href="/item">Item</Item>);

    expect(screen.getByRole('link', { name: 'Item' })).toHaveAttribute(
      'href',
      '/item',
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
