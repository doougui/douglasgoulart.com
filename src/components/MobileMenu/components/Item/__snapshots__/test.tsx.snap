// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> should render correctly 1`] = `
.c0 {
  padding-bottom: 2rem;
  padding-top: 2rem;
  border-bottom: 1px solid #504F57;
}

.c1 {
  color: #EDEDEF;
  font-size: 1.6rem;
  line-height: 1.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  gap: 7px;
}

<li
  class="c0"
>
  <a
    class="c1"
    href="/item"
  >
    Item
  </a>
</li>
`;
